packages/proxy-container/src/index.ts

```diff
@@ -0,0 +1,15 @@
+import { createServer } from 'node:http';
+import { proxyRequests } from '@discordjs/proxy';
+import { REST } from '@discordjs/rest';
+
+if (!process.env.DISCORD_TOKEN) {
+	throw new Error('A DISCORD_TOKEN env var is required');
+}
+
+// We want to let upstream handle retrying
+const api = new REST({ rejectOnRateLimit: () => true, retries: 0 }).setToken(process.env.DISCORD_TOKEN);
+// eslint-disable-next-line @typescript-eslint/no-misused-promises
+const server = createServer(proxyRequests(api));
```

Comment:
Where is the misused promise, and why can't we just `await` it, using TLA, which is supported in ESM?

Reply:
Misused promise is because `createServer` takes `(req, res) => void` and `proxyRequests` returns `(req, res) => Promise<void>`. Nothing we can await here (not that there is much of a point, given on those node versions uncaught promises like that will just crash the process)